src/main/java/com/demonwav/mcdev/insight/ListenerEventAnnotator.java

```diff
@@ -68,6 +60,38 @@ public void annotate(@NotNull PsiElement element, @NotNull AnnotationHolder hold
             return;
         }
 
+        PsiCodeBlock methodCodeBlock = method.getBody();
+        if (methodCodeBlock == null) {
+            return;
+        }
+        PsiStatement[] methodBodyStatements = methodCodeBlock.getStatements();
+        for (PsiStatement statement : methodBodyStatements) {
+
+            PsiAnnotation eventHandler = null;
+
+            for (AbstractModuleType<?> moduleType : moduleTypes) {
+                final List<String> listenerAnnotations = moduleType.getListenerAnnotations();
+                for (String listenerAnnotation : listenerAnnotations) {
+                    PsiAnnotation annotation = modifierList.findAnnotation(listenerAnnotation);
+                    if (annotation != null) {
+                        eventHandler = annotation;
+                    }
+                }
+            }
```

Comment:
Why is this loop done for ever `PsiStatement`, when it applies to the whole method?